src/libs/DistanceRequestUtils.ts

```diff
@@ -229,6 +229,25 @@ function getDistanceRequestAmount(distance: number, unit: Unit, rate: number): n
     return Math.round(roundedDistance * rate);
 }
 
+/**
+ * Extracts the distance from a merchant string.
+ *
+ * @param merchant - The merchant string containing the distance.
+ * @returns The distance extracted from the merchant string.
+ */
+function getDistanceFromMerchant(merchant: string | undefined, unit: Unit): number {
+    if (!merchant) {
+        return 0;
+    }
+
+    const distance = Number(merchant.split(' ')[0]);
+    if (!distance) {
+        return 0;
+    }
+
+    return unit === CONST.CUSTOM_UNITS.DISTANCE_UNIT_KILOMETERS ? distance / METERS_TO_KM : distance / METERS_TO_MILES;
```

Comment:
yes, but I would need to add another logic to replace 'km' or 'mi' with a proper translation string - we're operating everywhere on distance in meters. I just wanted to extract distance and meters and use another util which we're already using: `getDistanceForDisplay` (it accepts distance in meters)

Reply:
Oh I see, you're converting it back into meters here since you're dividing by the conversion factor.

NAB: But would you please add a comment explaining this. The whole thing is a bit convoluted 😅 but I think we both agree this is the lowest effort and most reasonable approach right now 👍

Reply:
yeah, of course, will do!